src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -247,6 +249,44 @@ Copyright (c) .NET Foundation. All rights reserved.
       <PackProjectInputFile>$(MSBuildProjectFullPath)</PackProjectInputFile>
     </PropertyGroup>
   </Target>
+  
+  <Target Name="_GetProjectReferenceVersions">    
+    <ConvertToAbsolutePath Paths="$(RestoreOutputPath)">
+      <Output TaskParameter="AbsolutePaths" PropertyName="RestoreOutputAbsolutePath" />
+    </ConvertToAbsolutePath>
+
+    <ConvertToAbsolutePath Condition="'$(ProjectAssetsFile)' != ''" Paths="$(ProjectAssetsFile)">
+      <Output TaskParameter="AbsolutePaths" PropertyName="ProjectAssetsFileAbsolutePath" />
+    </ConvertToAbsolutePath>
+
+    <GetProjectReferencesFromAssetsFileTask
+        RestoreOutputAbsolutePath="$(RestoreOutputAbsolutePath)"
+        ProjectAssetsFileAbsolutePath="$(ProjectAssetsFileAbsolutePath)">
+      <Output
+        TaskParameter="ProjectReferences"
+        ItemName="_ProjectReferencesFromAssetsFile" />
+    </GetProjectReferencesFromAssetsFileTask>
+
+    <Msbuild
+      Projects="@(_ProjectReferencesFromAssetsFile)"
+      Targets="_GetProjectVersion"
+      SkipNonexistentTargets="true"
```

Comment:
`SkipNonexistentProjects="true"`  should also be set, that probably would fail to build, but it still seems like it should be set